security/testdata/code.go

```diff
@@ -106,3 +106,32 @@ func SecureSecureWithOAuth2(ep goa.Endpoint, authOAuth2Fn security.AuthorizeOAut
 	}
 }
 `
+
+var SingleServiceAuthFuncsCode = `// AuthAPIKeyFn implements the authorization logic for APIKey scheme.
+func AuthAPIKeyFn(ctx context.Context, key string, s *security.APIKeyScheme) (context.Context, error) {
+	// Add authorization logic
+	if key == "" {
+		return ctx, fmt.Errorf("invalid key")
```

Comment:
I think this should return an instance of the error struct generated for unauthorized error.